torchvision/ops/feature_pyramid_network.py

```diff
@@ -87,25 +120,16 @@ def forward(self, x):
         names = list(x.keys())
         x = list(x.values())
 
-        last_inner = self.last_inner_block(x[-1])
+        last_inner = self.get_result_from_inner_blocks(x[-1], -1)
         results = []
-        results.append(self.last_layer_block(last_inner))
-
-        # TODO: rewrite as x[:-1][::-1] when supported in TS
-        x_except_last = x[0:len(x) - 1]
-        x_except_last.reverse()
-        out_dim_tensor = [x_out for x_out in x_except_last]
-
-        i = 0
-        for inner_block, layer_block in zip(self.inner_blocks, self.layer_blocks):
-            if i < len(out_dim_tensor):
-                feature = out_dim_tensor[i]
-                inner_lateral = inner_block(feature)
-                feat_shape = inner_lateral.shape[-2:]
-                inner_top_down = F.interpolate(last_inner, size=feat_shape, mode="nearest")
-                last_inner = inner_lateral + inner_top_down
-                results.insert(0, layer_block(last_inner))
-            i += 1
+        results.append(self.get_result_from_layer_blocks(last_inner, -1))
+
+        for idx in range(len(x) - 2, -1, -1):
+            inner_lateral = self.get_result_from_inner_blocks(x[idx], idx)
```

Comment:
Is it possible `len(inner_blocks)` < range(len(x) ? I think we need a check for that here

Reply:
yeah, we should check that and raise an error if that's not the case

Reply:
I'll do this in a follow-up PR, as all checks are green now and I'd like to get this merged ASAP